pkg/controller/platformadmin/util.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2023 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the License);
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an AS IS BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package platformadmin
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+
+	iotv1alpha2 "github.com/openyurtio/openyurt/pkg/apis/iot/v1alpha2"
+	"github.com/openyurtio/openyurt/pkg/controller/platformadmin/config"
+	utils "github.com/openyurtio/openyurt/pkg/controller/platformadmin/utils"
+)
+
+// NewYurtIoTDockComponent initialize the configuration of yurt-iot-dock component
+func NewYurtIoTDockComponent(platformAdmin *iotv1alpha2.PlatformAdmin, platformAdminFramework *PlatformAdminFramework) (*config.Component, error) {
+	var yurtIotDockComponent config.Component
+
+	// If the configuration of the yurt-iot-dock component that customized in the platformAdminFramework
+	for _, cp := range platformAdminFramework.Components {
+		if cp.Name != utils.IotDockName {
+			continue
+		}
+		return cp, nil
+	}
+
+	// Otherwise, the default configuration is used to start
+	ver, ns, err := utils.DefaultVersion(platformAdmin)
+	if err != nil {
+		return nil, err
+	}
+
+	yurtIotDockComponent.Name = utils.IotDockName
+	yurtIotDockComponent.Deployment = &appsv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"app":           "yurt-iot-dock",
+				"control-plane": "platformadmin-controller",
+			},
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"app":           "yurt-iot-dock",
+					"control-plane": "platformadmin-controller",
+				},
+				Namespace: ns,
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:            "yurt-iot-dock",
+						Image:           fmt.Sprintf("leoabyss/yurt-iot-dock:%s", ver),
```

Comment:
It would be better to replace "leoabyss" with "openyurt" to make it more appropriate.